### ISSUE TITLE
Legacy Tile Source fetching 2 layers

### ISSUE THREAD

**Issue by ec-dev-1** [openseadragon/openseadragon#2255]
Hello there,

Curious to know as to what could be causing the Legacy Tile Source to fetch 2 levels upon initializing.
This is happening on my implementation but can also be witnessed on the Legacy Tile Source example from this page:

https://openseadragon.github.io/examples/tilesource-legacy/

If you watch the network request, there are two images being fetched for the last example (house with red roof).

**Comment by iangilman** [comment]
Good question! In the case of the house with the red roof, I assume the lower res tile is for the navigator, and the higher res tile is for the viewer. Are you using a navigator in your version as well, or the reference strip?

**Comment by ec-dev-1** [comment]
@iangilman  thanks for responding.  I am not familiar with the navigator option.  Our implementation is very similar to the examples provided on the link I shared.  To summarize, we are providing 4 different 'levels' when initializing Open Sea Dragon.
The levels are in order of smallest to greatest, left to right.

Level 4 is the full resolution of the image.
In our implementation, OSD requests Levels 2 and 3.

I am uncertain if the request for multiple images is an error on our end, or if this is intended by OSD.

Can you link me to a documentation with references to the Navigator so I can confirm if this exists in our code?

**Comment by ec-dev-1** [comment]
@iangilman thanks for responding. I am not familiar with the navigator option. Our implementation is very similar to the examples provided on the link I shared. To summarize, we are providing 4 different 'levels' when initializing Open Sea Dragon.
The levels are in order of smallest to greatest, left to right.

Level 4 is the full resolution of the image.
In our implementation, OSD requests Levels 2 and 3.

I am uncertain if the request for multiple images is an error on our end, or if this is intended by OSD.

Can you link me to a documentation with references to the Navigator so I can confirm if this exists in our code?

**Comment by ec-dev-1** [comment]
@iangilman  thanks for that example.  I can confirm we do not use this feature.
Here's a public link to our implementation where you can see that OSD is fetching the 'quarter' and 'half' size images.
Looking forward to your response.

https://public.earthcam.net/earthcam/earthcampus/61_megapixelcam/view

thanks,

**Comment by iangilman** [comment]
Great, thank you! Cool site, BTW :) 

I think I've figured out what's going on... By default, OSD loads the smallest tile that covers the entire image first, and then loads higher resolution tiles afterwards. This makes a lot of sense with a tiled pyramid, because those higher resolution levels are composed of many tiles, and it may take a while for them all to load in. In your case, it's not as important, because all of your tile levels cover the entire image. 

If you watch closely, you can see the low resolution image show up briefly and then get replaced by the higher resolution one. It's not much of a delay between them, so you might as well just be loading the higher resolution one in your case. I believe you can make OSD do that by using `immediateRender: true` in your options when creating the viewer. 

It seems possible we should consider making that the default for `LegacyTileSource`. 🤔

**Comment by ec-dev-1** [comment]
@iangilman  thanks to your suggestion, I started going over each option that we added in our implementation to see what they do and was able to find that the property "homeFillsViewer" was causing the extra tile layer fetching since we are making sure that the image occupies the whole view and therefore requires a few steps of zooming-in.  "immediateRender" was already being assigned as true in our option.

Once again, thanks for looking into this with me!

I really appreciate your assistance.

Thanks!
